### ISSUE TITLE
Make all elements on screen dockable/movable/resizable

### ISSUE THREAD

**Issue by dacap** [aseprite/aseprite#518]
The color-bar, color-picker (#492), timeline, tabs (#118), minieditor (#404), and tool-bar should be dockable windows that can be moved to any other position on the screen.

Review #250, [t/933](https://community.aseprite.org/t/preview-on-another-window/933), [t/1065](https://community.aseprite.org/t/layer-selection-more-rapid/1065), [/5472](https://community.aseprite.org/t/customizing-layout-of-panels/5472)

Tasks:
- [x] Multiple native windows are required to drag-and-drop widgets to separated windows (#139)
- [x] Initial `Dock` widget to put widgets on each side, center, create subdocks ([preview](https://twitter.com/davidcapello/status/1440068577109766154))
- [x] Restore old layout configuration (color bar split pos, timeline pos, etc.) and migrate to new Dock layout
- [x] Load/saving Dock layout
- [x] Create & customize current layoout (drag-and-drop widgets, etc.)

Following work will be done in #5327

**Comment by SleepyEleven** [comment]
This feature will be the bee's knees when it is implemented. Can't wait.

**Comment by dacap** [comment]
Some possibilities requested by other people:

![cxgp0qpwqaa_lkg](https://cloud.githubusercontent.com/assets/39654/20245118/782ddeda-a978-11e6-845d-ee8a485e451f.jpg)
[Source](https://twitter.com/mckack/status/797605447939321857)

**Comment by dacap** [comment]
Related to #250

**Comment by Omar-AbdulAzeez** [comment]
something to consider for the future (if not already considered) is also making sprite tabs floating and dockable... if the palette, timeline, preview, (other sprite dependent elements) are already docked to that tab then they should undock with it with the exception being tools i.e.. only 1 tools element should exist at all times with the ability to hide it altogether from menubar > view... basically you can just leave it docked to the main window until specifically grabbed. Also don't forget to make copies of the menubar as well when undocking sprite tabs. if the sprite dependent elements are undocked they should be sprite tab specific... i.e.. should only dock back to the same sprite tab they came from and opening a new tab will still give it it's own docked elements even if there are undocked elements elsewhere... the sprite specific part could be an option toggle (FOR EACH ELEMENT SEPERATELY!) since it can be beneficial to just have those elements update themselves when focusing on another sprite (might create issues with selected cel and fg and bg color for each sprite tho idk)

this way it's essentially as if you're opening two instances of Aseprite while still benefiting from instance specific features like duplicate sprite view (oh yeah... a duplicate sprite view should not be considered a sprite tab but have the same relation to the original sprite tab as the preview window with the difference being that you can draw and use tools inside of it)

**Comment by dacap** [comment]
Hi @Omar-AbdulAzeez, I thought about the possibility to create new windows from tabs in the past + the complexities about the duplication of widgets (color-bar & timeline mainly) in the new window. At the moment there are no plans to implement that because there are too much code that depends on the main window + main widgets. But maybe in a far future we will be refactoring the code to support multiple "main windows" / floating tabs.

**Comment by HeyItsLollie** [comment]
I came here looking for word on making new windows from tabs, and I hope it does gain support in the future. The floating preview window has been a good step towards this, being able to separate elements of Aseprite onto different monitors helps free up window real-estate quite a bit, especially when working on a tablet.

I'd personally like to see Aseprite be able to instantiate a second program tab on the Windows Taskbar, so that it can operate like a true second window with mobile/duplicate widgets. It'd also be very useful for pairing with [ShaderGlass](https://github.com/mausimus/ShaderGlass), which can clone a specific window for applying shaders onto. (ShaderGlass currently can't do this with Aseprite's floating Preview window, but it's easy to maximize both the Preview window and ShaderGlass on a second monitor.)

**Comment by tjkirch** [comment]
One detail mentioned in https://github.com/aseprite/aseprite/issues/1585 but that I don't see here is the ability to dock the **undo history**.  That's something I like to have open all the time, and it'd be better if it were docked in a panel like the others shown above.

**Comment by ReversedField** [comment]
Hello I wanted to bump this just to show how important being able to move tools on the left side of the app would be to left-handed peoples. Thanks!

**Comment by nkalpakis21** [comment]
i see this has been open for awhile. is there trouble getting a dev to fix this issue? what is blocking this for so long

**Comment by thedyze** [comment]
Just having a single option in preferences to move the tool bar to the left would go a long way imo.

**Comment by mbrinton91** [comment]
It looks like this was closed because the feature was added? I can't figure out how to relocate the toolbar for the life of me though. Any documentation on this or am I missing something?

**Comment by Gasparoken** [comment]
Yes, but it was added in the beta version (i.e. [1.3.16-beta1](https://community.aseprite.org/t/aseprite-v1-3-15-3-v1-3-16-beta1/27469#p-51264-whats-new-1)) [currently only available through Steam](https://aseprite.org/beta)

**Comment by EnderNon** [comment]
> Yes, but it was added in the beta version (i.e. [1.3.16-beta1](https://community.aseprite.org/t/aseprite-v1-3-15-3-v1-3-16-beta1/27469#p-51264-whats-new-1)) [currently only available through Steam](https://aseprite.org/beta)

Can you not get this by compiling from src? I've been trying to figure out how to relocate this stuff, can't find it at all.

**Comment by v4u6h4n** [comment]
Is this available in v1.3.17 now? I don't know how to use it...

**Comment by EnderNon** [comment]
> Is this available in v1.3.17 now? I don't know how to use it...

No idea if it's in the current major release but I built from source and I have this button in the top right. Pressing it allows you to modify the layout using the white bars and save into presets.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/b073cf90-894a-4084-b9ee-764460c89fa9" />

**Comment by v4u6h4n** [comment]
@EnderNon Yeah I did find that feature, which is quite nice, I just couldn't figure out if I could dock floating windows, like the preview window. I think maybe not yet...?

**Comment by EnderNon** [comment]
> [@EnderNon](https://github.com/EnderNon) Yeah I did find that feature, which is quite nice, I just couldn't figure out if I could dock floating windows, like the preview window. I think maybe not yet...?

don't think you can unfortunately.

**Comment by v4u6h4n** [comment]
Okay no worries. Is it going to implemented? Or should I make a feature request?

**Comment by dacap** [comment]
> Okay no worries. Is it going to implemented? Or should I make a feature request?

It's planned but not yet implemented https://github.com/aseprite/aseprite/issues/5327
